huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go

```diff
@@ -0,0 +1,295 @@
+	}
+
+	authType := d.Get("auth_type").(string)
+	createOpts := engines.CreateOpts{
+		Payment:             "1",
+		SpecType:            "CSE2",
+		Name:                d.Get("name").(string),
+		Description:         d.Get("description").(string),
+		Flavor:              d.Get("flavor").(string),
+		AvailabilityZones:   utils.ExpandToStringList(d.Get("availability_zones").([]interface{})),
+		AuthType:            authType,
+		VpcName:             vpcResp.Name,
+		VpcId:               vpcResp.ID,
+		NetworkId:           networkId,
+		SubnetCidr:          subnetResp.CIDR,
+		PublicIpId:          d.Get("eip_id").(string),
+		Inputs:              d.Get("extend_params").(map[string]interface{}),
+		EnterpriseProjectId: common.GetEnterpriseProjectID(d, conf),
+	}
+
+	if authType == "RBAC" {
+		createOpts.AuthCred = &engines.AuthCred{
+			Password: d.Get("admin_pass").(string),
+		}
+	}
+
+	resp, err := engines.Create(client, createOpts)
+	if err != nil {
+		return diag.Errorf("error creating Microservice engine: %s", err)
+	}
+	d.SetId(resp.ID)
+
+	log.Printf("[DEBUG] Waiting for the Microservice engine to become running, the engine ID is %s.", d.Id())
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"Init", "Executing"},
+		Target:       []string{"Finished"},
+		Refresh:      MicroserviceJobRefreshFunc(client, d.Id(), strconv.Itoa(resp.JobId)),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        180 * time.Second,
+		PollInterval: 15 * time.Second,
+	}
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for the creation of Microservice engine (%s) to complete: %s", d.Id(), err)
+	}
+
+	return resourceMicroserviceEngineRead(ctx, d, meta)
+}
+
+func resourceMicroserviceEngineRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	client, err := conf.CseV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CSE v2 client: %s", err)
+	}
+
+	resp, err := engines.Get(client, d.Id(), common.GetEnterpriseProjectID(d, conf))
+	if err != nil {
+		return common.CheckDeletedDiag(d, parseEngineJobError(err), "error retrieving Microservice engine")
+	}
+
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("name", resp.Name),
+		d.Set("flavor", resp.Flavor),
+		d.Set("availability_zones", resp.Reference.AzList),
+		d.Set("auth_type", resp.AuthType),
+		d.Set("enterprise_project_id", resp.EnterpriseProjectId),
+		d.Set("network_id", resp.Reference.NetworkId),
+		d.Set("description", resp.Description),
+		d.Set("eip_id", resp.Reference.PublicIpId),
+		d.Set("extend_params", resp.Reference.Inputs),
+	)
+	// Attributes
+	if resp.Reference.ServiceLimit != "" {
+		limit, err := strconv.Atoi(resp.Reference.ServiceLimit)
+		if err != nil {
+			mErr = multierror.Append(mErr, fmt.Errorf("unable to parsing service limit (%#v)",
+				resp.Reference.ServiceLimit))
```

Comment:
can we add it into diags with warnning level or just ignore it?

Reply:
done.